core/data/src/main/java/com/mateuszholik/data/extensions/ApiModelsExtensions.kt

```diff
@@ -0,0 +1,133 @@
+package com.mateuszholik.data.extensions
+
+import com.mateuszholik.model.Competition
+import com.mateuszholik.model.CompetitionType
+import com.mateuszholik.model.Duration
+import com.mateuszholik.model.Group
+import com.mateuszholik.model.Head2Head
+import com.mateuszholik.model.Match
+import com.mateuszholik.model.MatchScore
+import com.mateuszholik.model.Referee
+import com.mateuszholik.model.RefereeType
+import com.mateuszholik.model.Score
+import com.mateuszholik.model.Season
+import com.mateuszholik.model.Stage
+import com.mateuszholik.model.Status
+import com.mateuszholik.model.Team
+import com.mateuszholik.model.TeamH2HData
+import com.mateuszholik.model.Winner
+import com.mateuszholik.network.models.AreaApi
+import com.mateuszholik.network.models.CompetitionApi
+import com.mateuszholik.network.models.Head2HeadApi
+import com.mateuszholik.network.models.MatchApi
+import com.mateuszholik.network.models.MatchScoreApi
+import com.mateuszholik.network.models.MatchesApi
+import com.mateuszholik.network.models.RefereeApi
+import com.mateuszholik.network.models.ScoreApi
+import com.mateuszholik.network.models.SeasonApi
+import com.mateuszholik.network.models.TeamApi
+import com.mateuszholik.network.models.TeamH2HDataApi
+
+internal fun MatchesApi.toCommonModel(): List<Match> =
+    matches.map { it.toCommonModel() }
+
+internal fun MatchApi.toCommonModel(): Match =
+    Match(
+        awayTeam = awayTeam.toCommonModel(),
+        competition = competition.toCommonModel(area),
+        group = getGroup(group),
+        homeTeam = homeTeam.toCommonModel(),
+        id = id,
+        lastUpdated = lastUpdated.toLocalDateTime(),
+        matchday = matchday,
+        referees = referees.map { it.toCommonModel() },
+        score = score.toCommonModel(),
+        season = season.toCommonModel(),
+        stage = Stage.valueOf(stage),
+        status = Status.valueOf(status),
+        utcDate = utcDate.toLocalDateTime(),
+        venue = venue.orEmpty()
+    )
+
+private fun getGroup(groupName: String?): Group =
+    when (groupName) {
+        "Group A" -> Group.GROUP_A
```

Comment:
you can put these names in enum class 
`enum class Group(internal val groupName: String)`
and resolve it like 
`Group.values().firstOrNull { it.groupName == groupName } ?: Group.GROUP_NA`